src/VecSim/vec_sim.cpp

```diff
@@ -46,6 +46,12 @@ extern "C" double VecSimIndex_GetDistanceFrom(VecSimIndex *index, size_t id, con
     return index->getDistanceFrom(id, blob);
 }
 
+extern "C" void VecSim_Normalize(void *blob, size_t dim, VecSimType type) {
```

Comment:
do you assume the buffer is mutable?

Reply:
Yes. either send the original vector or make a copy yourself

Reply:
although I can move `VecSimType_sizeof` to the library and then with the same args have the copy part also